src/app/content/print.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { GoogleAnalyticsClient } from '../../gateways/googleAnalyticsClient';
+import googleAnalyticsClient from '../../gateways/googleAnalyticsClient';
+import { assertWindow } from '../utils';
+import { print } from './print';
+import { Book } from './types';
+
+describe('print', () => {
+  let client: GoogleAnalyticsClient;
+  let mockGa: any;
+  let mockPrint: any;
+  const window: Window = assertWindow();
+  let originalLocation: any;
+  const pathname: string = '/books/biology-2e/pages/1-review-questions';
+  const book = { id: 'bookId', slug: 'biology-2e', content: 'fooobarcontent' } as any as Book;
+
+  beforeEach(() => {
+    client = googleAnalyticsClient;
+    mockPrint = jest.fn();
+    window.print = mockPrint;
+    mockGa = jest.fn<UniversalAnalytics.ga, []>();
+    window.ga = mockGa;
+
+    originalLocation = window.location;
+    delete window.location;
+
+    client.setTrackingIds(['foo']);
+    jest.resetAllMocks();
+  });
```

Comment:
this tests leaks globals for `window.ga` and `window.print` you should back up the original values and restore them in an `afterEach` like this:

```js
const printBack = window.print;
beforeEach(() => {
  window.print = mockPrint;
});
afterEach(() => {
  window.print = printBack;
})
```

and for the ga too